### ISSUE TITLE
Some XML Files are gigantic since 3.15 update and load times are awful (EditedNode items)

### ISSUE THREAD

**Issue by lutomlin** [PathOfBuildingCommunity/PathOfBuilding#3156]
### Check version

- [X] I'm running the latest version of Path of Building and I've verified this by checking the [changelog](https://github.com/PathOfBuildingCommunity/PathOfBuilding/blob/master/CHANGELOG.md)

### Check for duplicates

- [X] I've checked for duplicate issues by using the search function of the [issue tracker](https://github.com/PathOfBuildingCommunity/PathOfBuilding/issues)

### What is the expected behaviour/value?

Fast/responsive load times when browsing folders at the main build list menu. Fast loading once you start to load a build.

### What is the actual behaviour/value?

Browsing to my folders with builds made since 3.15 is very, very slow.

I've noticed that the file size for build XML files has gotten absolutely enormous, going from tens of kilobytes to over 3MB in my worst-case.

This seems to be due to the new Timeless jewel editing functionality. I noticed that almost the whole XML file is composed of `EditedNode` items. I've attached a pastebin.

### How to reproduce the issue

Make builds with lots of timeless jewels. Attempt to browse to a folder containing their XMLs in PoB. Experience slowness.

I've noticed that these file sizes seem to persist between XMLs, probably because of the timeless jewel node edit persistence introduced in the 3.15 update. Making a new blank build with just one skill gem equipped and saving it creates an XML file of over 1MB in size now.

### Build code

```shell
https://pastebin.com/90XfZ141 is a dump of the worst offending build (3.65MB XML on disk).
```

**Comment by Wires77** [comment]
Yeah, I noticed this could be an issue here: https://github.com/PathOfBuildingCommunity/PathOfBuilding/issues/2943#issuecomment-884886035

The editedNode persistence is partially a feature, as if we ever upload builds to a central website, we could potentially harvest that node information from those builds.  On the other hand, the file size is a real problem, so it should probably be cleared whenever you exit a build/start a new one

**Comment by lutomlin** [comment]
Looks like that helps - the fresh file is only 13kb instead of 1.1mb. It's still a problem though - I have 30+ builds that have a huge file at the moment and the problem will only reappear even if I do try and fix them.

This is probably out in the wild affecting a lot of people at the moment as well as me. Do you think you could disable it by default or something? It's not a feature I understand or particularly want and it's degrading my experience quite badly with long load times of like 5+s just to pull up my 3.15 builds folder each time I want to look at something.

Also - how do I actually control this feature? Do I have to set the timeless jewel seed appropriately before I start applying node changes, and then the same seed in a different PoB will load the same node changes? I usually don't set the node changes to be for a specific seed id, I usually add a default-seed jewel and set a couple of them to nodes I would like generally in order to test out what nodes I should be searching for when I look to roll/purchase a jewel. If *that* stuff is being copied from build to build, then that is probably more like a bug than a feature. In fact, loading up one of my PoBs with a Brutal Restraint, it seems like it's copying nodes a bit wildly, a whole bunch of nodes have 5% Dex added to them when I certainly didn't apply that in this file (and I don't remember doing it at all for those nodes).

**Comment by Wires77** [comment]
> This is probably out in the wild affecting a lot of people at the moment as well as me. Do you think you could disable it by default or something? It's not a feature I understand or particularly want and it's degrading my experience quite badly with long load times of like 5+s just to pull up my 3.15 builds folder each time I want to look at something.

Yeah, it'll definitely be fixed, thanks for bringing it up.

> Also - how do I actually control this feature? Do I have to set the timeless jewel seed appropriately before I start applying node changes, and then the same seed in a different PoB will load the same node changes? I usually don't set the node changes to be for a specific seed id, I usually add a default-seed jewel and set a couple of them to nodes I would like generally in order to test out what nodes I should be searching for when I look to roll/purchase a jewel. If _that_ stuff is being copied from build to build, then that is probably more like a bug than a feature. In fact, loading up one of my PoBs with a Brutal Restraint, it seems like it's copying nodes a bit wildly, a whole bunch of nodes have 5% Dex added to them when I certainly didn't apply that in this file (and I don't remember doing it at all for those nodes).

Yeah, Timeless jewels will always be the same if they have the same seed.  So the idea is if you happen transfer the timeless jewel with the same seed to a different character, you won't have to redo all of the edited nodes.  Since you're using the same default seed when creating characters, they'll all switch their nodes to whatever you had edited it to on a different character.

This cross-character functionality is only really used here, so it's a bit strange, but I was optimistic about having a central build website other than pastebin that could parse those build XMLs and start actually getting good data for timeless jewel seeds.

**Comment by lutomlin** [comment]
I imagined this a while ago and designed a storage system that could handle it. It wouldn't just take nodes from PoB XMLs though - it's too error-prone, people would just put in random nodes when trying to figure out their build and figure out what they wanted etc. You'd have to have a submissions process, a video proof system, that kind of thing, so that people don't get scammed trying to buy a jewel they think is good but is actually just someone's random god-tier wish jewel.

**Comment by Wires77** [comment]
Yeah, the thought was people aren't going to go through the effort to do all that, so you'll hardly have any data.  With this, people would import their characters, edit the nodes they care about, then upload them again to share.  You'll have some bad data for sure, but that would mostly be on the default seed

**Comment by lutomlin** [comment]
Yeah, for sure it would have to be a community collaboration and not just a data source :)

Thanks for taking the time to respond and fix this!
